flow-typed/redux.js

```diff
@@ -0,0 +1,74 @@
+declare module 'redux' {
+
+  /*
+
+    S = State
+    A = Action
+
+  */
+
+  declare type Dispatch<S, A> = (action: A) => A;
```

Comment:
I can't think of anything better at the moment, since any middleware could modify this in all sorts of ways. It might be possible to pass through the definition of `Dispatch` from `StoreEnhancer`. However, I think I remember reading that a reason Flow typedefs for Redux were put on hold last year was because this was difficult to achieve.